### ISSUE TITLE
Cannot reproduce results for Centernet-104

### ISSUE THREAD

**Issue by Michaelyq** [Duankaiwen/CenterNet#75]
I tried both CenterNet-52 and CenterNet-104 on the COCO dataset using the default settings (e.g., 480,000 iterations). CenterNet-52 achieves mAPs of 0.416 and 0.433 respectively, which is close to the results reported in the original paper. But for the CenterNet-104 using Hourglass-104 as the backbone, I only got mAP of 0.268. 

I used "instances_val2017.json" as the test set.

Could anybody help me?

The following the configurations and results for CenterNet-104 at iteration of 480,000.
-------------------------------------------------------------------------------------------
cfg_file: config/CenterNet-104.json
loading all datasets...
split: val
loading from cache file: cache/coco_val2017.pkl
loading annotations into memory...
Done (t=0.66s)
creating index...
index created!
system config...
{'batch_size': 16,
 'cache_dir': 'cache',
 'chunk_sizes': [2, 2, 2, 2, 2, 2, 2, 2],
 'config_dir': 'config',
 'data_dir': '/dataset/public',
 'data_rng': <mtrand.RandomState object at 0x7f1ff1e8aa68>,
 'dataset': 'MSCOCO',
 'decay_rate': 10,
 'display': 5,
 'learning_rate': 0.00025,
 'max_iter': 480000,
 'nnet_rng': <mtrand.RandomState object at 0x7f1ff1e8aab0>,
 'opt_algo': 'adam',
 'prefetch_size': 6,
 'pretrain': None,
 'result_dir': 'results',
 'sampling_function': 'kp_detection',
 'snapshot': 5000,
 'snapshot_name': 'CenterNet-104',
 'stepsize': 450000,
 'test_split': 'testdev2017',
 'train_split': 'train',
 'val_iter': 500,
 'val_split': 'val',
 'weight_decay': False,
 'weight_decay_rate': 1e-05,
 'weight_decay_type': 'l2'}
db config...
{'ae_threshold': 0.5,
 'border': 128,
 'categories': 80,
 'data_aug': True,
 'gaussian_bump': True,
 'gaussian_iou': 0.7,
 'gaussian_radius': -1,
 'input_size': [511, 511],
 'kp_categories': 1,
 'lighting': True,
 'max_per_image': 100,
 'merge_bbox': False,
 'nms_algorithm': 'exp_soft_nms',
 'nms_kernel': 3,
 'nms_threshold': 0.5,
 'output_sizes': [[128, 128]],
 'rand_color': True,
 'rand_crop': True,
 'rand_pushes': False,
 'rand_samples': False,
 'rand_scale_max': 1.4,
 'rand_scale_min': 0.6,
 'rand_scale_step': 0.1,
 'rand_scales': array([0.6, 0.7, 0.8, 0.9, 1. , 1.1, 1.2, 1.3]),
 'special_crop': False,
 'test_scales': [1],
 'top_k': 70,
 'weight_exp': 8}
loading parameters at iteration: 480000
building neural network...
module_file: models.CenterNet-104
total parameters: 210062960
loading parameters...
loading model from cache/nnet/CenterNet-104/CenterNet-104_480000.pkl
locating kps:   0%|                                    | 0/5000 [00:00<?, ?it/s]/usr/local/conda/lib/python3.7/site-packages/torch/nn/modules/upsampling.py:122: UserWarning: nn.Upsampling is deprecated. Use nn.functional.interpolate instead.
  warnings.warn("nn.Upsampling is deprecated. Use nn.functional.interpolate instead.")
locating kps: 100%|█████████████████████████| 5000/5000 [32:02<00:00,  2.68it/s]
Loading and preparing results...
DONE (t=3.33s)
creating index...
index created!
Running per image evaluation...
Evaluate annotation type *bbox*
DONE (t=75.56s).
Accumulating evaluation results...
DONE (t=12.13s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.268
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.401
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.279
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.133
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.285
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.366
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.256
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.402
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.436
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.228
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.474
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.596
Running per image evaluation...
Evaluate annotation type *bbox*
DONE (t=72.17s).
Accumulating evaluation results...
DONE (t=13.59s).
 False discovery    (FD) @[ IoU=0.05:0.50 | area=   all | maxDets=100 ] = 0.543
 False discovery    (FD) @[ IoU=0.05      | area=   all | maxDets=100 ] = 0.495
 False discovery    (FD) @[ IoU=0.25      | area=   all | maxDets=100 ] = 0.535
 False discovery    (FD) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.599
 False discovery    (FD) @[ IoU=0.05:0.50 | area= small | maxDets=100 ] = 0.715
 False discovery    (FD) @[ IoU=0.05:0.50 | area=medium | maxDets=100 ] = 0.511
 False discovery    (FD) @[ IoU=0.05:0.50 | area= large | maxDets=100 ] = 0.455

**Comment by Duankaiwen** [comment]
@Michaelyq Hi, i think the batch size may be too small

**Comment by Duankaiwen** [comment]
Can I have a look at your log?

**Comment by Michaelyq** [comment]
@Duankaiwen I trained CenterNet-104 on 8 TITAN_xp GPUs, each of which has 12 GB memory capacity that can only be allocated with 2 images (per/GPU). I'm wondering what is your setting of batch size for training CenterNet-104? Will different batch size settings affect the final performance of your model to a large amount?

Here is part of my training log for CenterNet-104. 
[centernet-from-245000-iters.txt](https://github.com/Duankaiwen/CenterNet/files/3449609/centernet-from-245000-iters.txt)

**Comment by Duankaiwen** [comment]
@Michaelyq The training loss is still too large. I use V100 with 32GB memory, which can be allocated  with 6 batches per GPU. If you don't have large memory, try loading the final model of Cornernet as the pretrained model and finetune it, i'm not sure, maybe it will work.

**Comment by apatsekin** [comment]
Same issue here. After 120k iterations on MS COCO training loss is around 5. 
[2,2,2,2] batch. @Duankaiwen could you provide what is the target loss for paper results?

**Comment by Duankaiwen** [comment]
@apatsekin sorry, i didn't save the full training log. The final loss is about :
training loss: 3.5885119438171387
validation loss at iteration: 3.860023260116577

**Comment by JoJoliking** [comment]
@Michaelyq @Duankaiwen  Have you solve the problem? I also set the batchsize in 16 in four 1080TI. And What value should the learning rate be set?
